.spelling

```diff
@@ -0,0 +1,147 @@
+# markdown-spellcheck spelling configuration file
+# Format - lines beginning # are comments
+# global dictionary is at the start, file overrides afterwards
+# one word per line, to define a file override use ' - filename'
+# where filename is relative to this configuration file
+AlertManager
+Ansible
+CentOS
+CloudFront
+CockroachDB
+Conda
+CoreOS
+Couchbase
+Crunchbase
+D2iQ
+Datastax
+Dex
+Dispatchfile
+Dockerfile
+ESXi
+ElasticSearch
+Flagger
+Flink
+GitOps
+Github
+Golang
+Grafana
+HAProxy
+Hadoop
+Hashicorp
+Horovod
+Iinc
+InfluxData
+Istio
+Jupyter
+JupyterHub
+JupyterLab
+K8s
+Katib
+Kerberos
+Kibana
+Knative
+Kommander
+Konvoy
+Konvoy
+Kubeflow
+Kubernetes
+Linkerd
+Logstash
+MXNet
+MacBook
+MacOS
+Mesos
+MongoDB
+MySQL
+Nagios
+Nginx
+NiFi
+OneLogin
+ReLU
+OpenID
+OpenResty
+Percona
+PerconaXtraDB
+PipelineRuns
+PostgreSQL
+PyTorch
+Segment.io
+Splunk
+Tekton
+Telegraf
+TensorFlow
+TitanDB
+Toree
+VMWare
+VMware
+VirtualService
+XtraDB
+addon
+artifact
+artifacts
+autoscaler
+config
+configs
+hostname
+kubeaddon
+kubeaddons
+lifecycle
+mTLS
+plugin
+spec
+hyperparameter
+hyperparameters
+serverless
+SQLite
+KFServing
+MLOps
+MinIO
+MutatingWebhookConfiguration
+base64
+gRPC
+
+#
+# Stuff needed to make Kubeflow error free
+#
+28x28
+3rd-party
+ConfigMaps
+Eco-system
+# Should be ElasticSearch?
```

Comment:
does the spelling tool mark this correct when `Search` is capitalized or you'd have to add a rule for it in either case?

Reply:
hunspell (the engine that's used under the hood of almost every open source checker) seems to go with smart-casing. so if we add `elasticsearch`, all things will be valid. but if we add `ElasticSearch`, then `Elasticsearch` will still be invalid.

Reply:
probably we should not go with all lower since you don't want the tool to mistakenly allow spellings like `ElaSticsearch`